### PR TITLE
Support S3 Private Link

### PR DESCRIPTION
**Pull request by legiangthanh**
In the current code, If the endpoint is S3 Private Link (e.g: bucket.vpce-1a2b3c4d-5e6f.s3.us-east-1.vpce.amazonaws.com), The plugin can't connect to S3 with below error message
` Message: The authorization header is malformed; the region 'vpce' is wrong; expecting 'us-east-1' (Service: Amazon S3; Status Code: 400; Error Code: AuthorizationHeaderMalformed; Request ID: 4SJFB38TBFE5CP1M; S3 Extended Request ID: lPPhAhTyC9aQO9bF0XMa5wqzaFpVO927JJRexs4HZjj0gzjOeA9rELX++mVWCkmM9kHD2DNhvJE=)`

This PR fixes this issue by enabling `ForceGlobalBucketAccessEnabled`, which will force the SDK to use the region in the exception. (Please refer details in: https://github.com/aws/aws-sdk-java/issues/1107#issuecomment-293022788)

### CODE REVIEW

**Review by minidragon88 — APPROVED** [review]
LGTM

JFYI for other reviewers `https://github.com/aws/aws-sdk-java/issues/1107#issuecomment-293022788`

File: src/main/java/org/embulk/input/s3/S3FileInputPlugin.java [hunk 1]
```diff
@@ -198,7 +198,7 @@ public TransactionalFileInput open(TaskSource taskSource, int taskIndex)
         return new S3FileInput(task, taskIndex);
     }
 
-    private AWSCredentialsProvider getCredentialsProvider(PluginTask task)
```
Comment (dmikurube) [inline src/main/java/org/embulk/input/s3/S3FileInputPlugin.java:201]: Understood `withForceGlobalBucketAccessEnabled`, but why did you need to change `private` to `protected`?

Comment (legiangthanh) [reply src/main/java/org/embulk/input/s3/S3FileInputPlugin.java:201]: In the current code, we extend this class and re-implement getCredentialsProvider to add some validation checks.

Comment (legiangthanh) [reply src/main/java/org/embulk/input/s3/S3FileInputPlugin.java:201]: I mean in the TD implementation When we use version 0.5.3 `getCredentialsProvider` is protected, we override this function to add some validation and remove some unused Provider base on our use case.

Comment (dmikurube) [reply src/main/java/org/embulk/input/s3/S3FileInputPlugin.java:201]: * That's TD's internal reason, and this is an OSS repository. Do not bring TD's internal reasons here. The OSS repo does not have reasons to accept TD's internal things.
* That's totally unrelated to "Support S3 Private Link". Do not mix up a different topic in a single PR.

Comment (dmikurube) [reply src/main/java/org/embulk/input/s3/S3FileInputPlugin.java:201]: If it is really needed, create another PR just for making it `protected`, and give a reasonable description why that is useful even for the OSS repo.

Comment (legiangthanh) [reply src/main/java/org/embulk/input/s3/S3FileInputPlugin.java]: Thanks, I revert the change back, let me make another PR for changing it to `protected`.

**Review by dmikurube — APPROVED** [review]
Then it's good, and approved. But could you squash the commits?

**Comment by dmikurube** [comment]
Merged. Thanks!
